python/mirascope/llm/clients/anthropic/_utils/encode.py

```diff
@@ -56,8 +56,66 @@ class MessageCreateKwargs(TypedDict, total=False):
 def _encode_content(
     content: Sequence[ContentPart], encode_thoughts: bool
 ) -> str | Sequence[anthropic_types.ContentBlockParam]:
-    """Convert mirascope content to Anthropic content format."""
+    """Returns encoded Anthropic content blocks from mirascope content."""
+    if len(content) == 1 and content[0].type == "text":
+        return content[0].text
+
+    blocks: list[anthropic_types.ContentBlockParam] = []
+
+    for part in content:
+        if part.type == "text":
+            blocks.append(anthropic_types.TextBlockParam(type="text", text=part.text))
+        elif part.type == "image":
+            source: (
+                anthropic_types.Base64ImageSourceParam
+                | anthropic_types.URLImageSourceParam
+            )
+            if part.source.type == "base64_image_source":
+                source = anthropic_types.Base64ImageSourceParam(
+                    type="base64",
+                    media_type=encode_image_mime_type(part.source.mime_type),
+                    data=part.source.data,
+                )
+            else:
+                source = anthropic_types.URLImageSourceParam(
+                    type="url",
+                    url=part.source.url,
+                )
+            blocks.append(anthropic_types.ImageBlockParam(type="image", source=source))
+        elif part.type == "tool_output":
+            blocks.append(
+                anthropic_types.ToolResultBlockParam(
+                    type="tool_result",
+                    tool_use_id=part.id,
+                    content=str(part.value),
+                )
+            )
+        elif part.type == "tool_call":
+            blocks.append(
+                anthropic_types.ToolUseBlockParam(
+                    type="tool_use",
+                    id=part.id,
+                    name=part.name,
+                    input=json.loads(part.args),
+                )
+            )
+        elif part.type == "thought":
+            if encode_thoughts:
+                blocks.append(
+                    anthropic_types.TextBlockParam(
+                        type="text", text="**Thinking:** " + part.thought
+                    )
+                )
+        else:
+            raise NotImplementedError(f"Content type {part.type} not supported")
 
+    return blocks
+
+
+async def _encode_content_async(
```

Comment:
I don't understand why we would duplicate this code to have an async version? Can you explain this change?

Reply:
The async versions are still here?